src/public/videoApp.ts

```diff
@@ -0,0 +1,169 @@
+import { sendMessageToParent } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { FrameContexts } from './constants';
+/**
+ * Namespace to video extensibility of the SDK.
+ *
+ */
+export namespace videoApp {
+  /**
+   * Represents a video frame.
+   */
+  export interface VideoFrame {
+    /**
+     * Video frame width.
+     */
+    width: number;
+    /**
+     * Video frame height.
+     */
+    height: number;
+    /**
+     * Video frame buffer
+     */
+    data: Uint8ClampedArray;
+    /**
+     * NV12 luma stride, valid only when video frame format is NV12
+     */
+    lumaStride: number;
+    /**
+     * NV12 chroma stride, valid only when video frame format is NV12
+     */
+    chromaStride: number;
+    /**
+     * RGB stride, valid only when video frame format is RGB
+     */
+    stride: number;
+  }
+  /**
+   * Video frame format enum
+   */
+  export enum VideoFrameFormat {
+    RGB,
+    NV12,
+  }
+  /**
+   * Video frame configuration supplied to Teams to customize the generated video frame parameters, like format.
+   */
+  export interface VideoFrameConfig {
+    /**
+     * video format
+     */
+    format: VideoFrameFormat;
+  }
+
+  /**
+   *  Video effect change type enum
+   */
+  export enum EffectChangeType {
+    /**
+     * current video effect changed.
+     */
+    EffectChanged,
+    /**
+     * disable the video effect
+     */
+    EffectDisabled,
+  }
+
+  /**
+   *  Video frame call back funtion definition
+   */
+  type VideoFrameCallback = (
+    frame: VideoFrame,
+    notifyVideoFrameProcessed: () => void,
+    notifyError: (errorMessage: string) => void,
+  ) => void;
+
+  /**
+   *  Video effect change call back funtion definition
+   */
+  type VideoEffectCallBack = (effectName: string | undefined) => void;
+
+  /**
+   * VideoApp
+   */
+  class VideoApp {
+    private videoFrameCallback: VideoFrameCallback;
+    private videoEffectCallback: VideoEffectCallBack;
+    /**
+     * register to read the video frames in Permissions section.
+     */
+    public registerForVideoFrame(frameCallback: VideoFrameCallback, config: VideoFrameConfig): void {
+      ensureInitialized(FrameContexts.sidePanel, FrameContexts.meetingStage);
```

Comment:
Are 'sidePanel' and 'meetingStage' correct frame context for video extensibility?